app/controllers/api/v3/sync_controller.rb

```diff
@@ -11,12 +11,14 @@ def __sync_from_user__(params)
   end
 
   def __sync_to_user__(response_key)
-    AuditLog.create_logs_async(current_user, records_to_sync, "fetch", Time.current) unless disable_audit_logs?
+    records = records_to_sync
+
+    AuditLog.create_logs_async(current_user, records, "fetch", Time.current) unless disable_audit_logs?
```

Comment:
Can we remove this ? If we still need the logging, we could send it to Datadog I guess, but its adding overhead to the sync path here:

```
    records_by_class = records.group_by { |record| record.class.to_s }

    records_by_class.each do |record_class, records_for_class|
      log_data = {
        user_id: user.id,
        record_class: record_class,
        record_ids: records_for_class.map(&:id),
        action: action,
        time: time
      }.to_json
```

Note that the above is *not* done async, its done before the handoff to the sidekiq job.

If no-one has looked at these logs in the past month or so, maybe we can just drop them.

Reply:
We debated this a fair bit, and yes that block of code is synchronous. No one has looked at them not just in the past month or so, but ever, in the last 3 years.

However, the argument for keeping this is a fail-safe I think, it's like insurance.

I'll post some numbers shortly around all of these, which will show that the time consumed by this is relatively small to the other things, which is why we didn't bother cleaning this up.

Reply:
We can't drop or move these to datadog afaik.

The purpose of this is to be able to audit which users have accessed specific data. 
The example scenario we've used for this is: _a legal authority asks us for information on which users have accessed data related to x._

Unless something has changed so that this is no longer a scenario we need to provide logs for, we'll need to keep this around in some form. Datadog doesn't work because we only retain data for 30 days there.

Reply:
Correcto, this is a compliance issue that we need to be bullet-proof against. We'll have to take the perf hit for it. That being said, if there is any data massaging that we can push into an async job, all the better.

Reply:
![image](https://user-images.githubusercontent.com/16774200/102588070-d903b080-4132-11eb-8863-912730765f64.png)

Fwiw this takes _extremely_ little time (only ~1% of the response time, ~20ms). I think we can keep this around, even without moving out the data massaging and come back to this later in a fun friday cleanup.